modules/dnn/src/layers/convolution_arm/fast_conv_2d.hpp

```diff
@@ -0,0 +1,699 @@
+                            int xi = xi_ + xytab[k*2+1];
+                            if ((unsigned)yi < (unsigned)Hi &&
+                                (unsigned)xi < (unsigned)Wi)
+                            {
+                                const float* inptr = inp + inp_plane_ofs + yi*Wi + xi;
+                                for (int c = 0; c < Cg; c++, inptr += Hi*Wi)
+                                {
+                                    //if (y0 == 0 && x0 == 0 && yi == 0 && xi == 0)
+                                    //    printf("c == %d: inpval = %.2f\n", c, *inptr);
+                                    inpbuf[c*(Hk*Wk*FX_CONV_NR)] = *inptr;
+                                }
+                            }
+                            else
+                            {
+                                for (int c = 0; c < Cg; c++)
+                                    inpbuf[c*(Hk*Wk*FX_CONV_NR)] = 0.f;
+                            }
+                        }
+                    }
+                    if (++x0 >= W0)
+                    {
+                        x0 = 0;
+                        ++y0;
+                    }
+                }
+            }
+
+            // 2. do convolution, compute Kg x (yx1 - yx0) part of the output tensor
+            {
+                int outstep0 = H0*W0;
+                float* outptr0 = out + (n*ngroups + g)*Kg*outstep0 + yx0;
+                // n * ngroups -> batch size, g is in this batch, kg is number of channel in this group, yx0 is the begin of this matrix.
+                float cbuf[FX_CONV_MR*FX_CONV_NR];
+                bool partial0 = yx1 - yx0 < FX_CONV_NR;
+                for(int k = 0; k < Kg; k += FX_CONV_MR, outptr0 += outstep0*FX_CONV_MR)
+                {
+                    int dk = Kg - k < FX_CONV_MR ? Kg - k : FX_CONV_MR;
+                    bool partial = partial0 || dk < FX_CONV_MR;
+                    float* outptr = outptr0;
+                    int outstep = outstep0;
+                    if (partial)
+                    {
+                        outptr = cbuf;
+                        outstep = FX_CONV_NR;
+                    }
+                    conv_block(HkWkCg, conv->weightsPtr+(g*Kg_aligned + k)*HkWkCg,
+                               inpbuf_task, outptr, outstep, conv->biasPtr + Kg*g + k,
+                               minval, maxval, conv->ifMinMaxAct);
+
+                    if (partial)
+                    {
+                        for (int k1 = 0; k1 < dk; k1++)
+                        {
+                            memcpy(outptr0 + k1*outstep0, &cbuf[k1*FX_CONV_NR],
+                                   (yx1 - yx0)*sizeof(cbuf[0]));
+                        }
+
+                    }
+                }
+            }
+
+        }
+    }
+
+    free(inpbuf_all);
+
+    // Activation
+    {
+        int outstep0 = H0*W0;
+        if( conv->activ )
+        {
+#ifdef _OPENMP
+#pragma omp parallel for
+#endif
```

Comment:
don't make activation a separate parallel loop, put it inside a single parallel loop